From 079cae9d7628f16b9efdf5723d2bfae5bdc2ba4e Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Tue, 27 Jan 2015 18:27:32 -0300
Subject: [PATCH] Make c_rehash match commands starting with - (minus) instead
 of minus in any starting position, otherwise a directory named a-b breaks it

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 tools/c_rehash.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/c_rehash.in b/tools/c_rehash.in
index 887e927..1df2fab 100644
--- a/tools/c_rehash.in
+++ b/tools/c_rehash.in
@@ -15,7 +15,7 @@ my $symlink_exists=eval {symlink("",""); 1};
 my $removelinks = 1;
 
 ##  Parse flags.
-while ( $ARGV[0] =~ '-.*' ) {
+while ( $ARGV[0] =~ '^-.*' ) {
     my $flag = shift @ARGV;
     last if ( $flag eq '--');
     if ( $flag =~ /-old/) {
-- 
2.0.5

